twitterpuller/src/main/java/de/brandwatch/minianalytics/twitterpuller/twitter/TweetFetcher.java

```diff
@@ -1,51 +1,24 @@
 package de.brandwatch.minianalytics.twitterpuller.twitter;
 
-import de.brandwatch.minianalytics.twitterpuller.kafka.Producer;
-import de.brandwatch.minianalytics.twitterpuller.model.Resource;
-import twitter4j.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import twitter4j.TwitterStream;
+import twitter4j.TwitterStreamFactory;
 
-import java.time.Instant;
 
+@Component
 public class TweetFetcher {
 
-    public TwitterStream getTwitterStream(Producer producer) {
+    private final TwitterPullerStatusListener listener;
 
+    @Autowired
+    public TweetFetcher(TwitterPullerStatusListener listener) {
+        this.listener = listener;
+    }
 
-        TwitterStream twitterStream = new TwitterStreamFactory().getInstance();
-        twitterStream.addListener(new StatusListener() {
-            public void onStatus(Status status) {
-                Instant date = Instant.ofEpochMilli(status.getCreatedAt().getTime());
-
-                Resource resource = new Resource();
-                resource.setDate(date);
-                resource.setText(status.getText());
-                resource.setAuthor(status.getUser().getName());
-
-                //Post mention into Kafka Topic
-                producer.send(resource);
-
-            }
-
-            public void onDeletionNotice(StatusDeletionNotice statusDeletionNotice) {
-            }
-
-            public void onTrackLimitationNotice(int i) {
-
-            }
-
-            public void onScrubGeo(long l, long l1) {
-
-            }
-
-            public void onStallWarning(StallWarning stallWarning) {
-
-            }
-
-            public void onException(Exception e) {
-                e.printStackTrace();
-            }
-        });
-
+    public TwitterStream getTwitterStream() {
```

Comment:
Looking at it like this, now it would be even more straightforward to define as a simple bean, no?

```java
@Bean
public twitterStream(TwitterPullerStatusListener listener){
    TwitterStream twitterStream = TwitterStreamFactory.getSingleton();
    twitterStream.addListener(listener);
    return twitterStream;
}

Then you wouldn't need this class at all.